### PR TITLE
Implemented "Track" Feat

### LINKED ISSUE #31: [FEAT] Task
A task is a piece of work or exercise to be done or undertaken by the mentees inline with the various stages they opt in for.

**The task feature should have the following:**

1. Track - Users/Mentees should be able to select tasks based on his/her track (frontend, backend or mobile)
2. Stage - tasks should be allocated to Mentees/Users according to their stage (newbie, amateur, pro).
3. Title - title of the task/exercise to be performed.
4. Description - An explanation of the what the exercise entails.
5. Deadline - There should be a deadline for the submission of a particular exercise. If user exceeds deadline, user will start afresh.
6. Status - Mentee/User should be able to pick up from where he/she stopped provided he/she has not approached deadline. Provision for draft and submit feature should be created .

### LINKED ISSUE #34: Send confirmation email endpoint Internal server Error
The endpoint gives a 500 error when I tried it out.

### PR DESCRIPTION
**Pull request by ofuochi**
## Description
It implements the 'Track' feature which closes #31 and resolves #34

## How Has This Been Tested?
Unit and integration tests have been written to test these features

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by tayormi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692&bid=18712017)

File: test/jest-e2e.json [hunk 1]
```diff
@@ -1,9 +1,16 @@
 {
   "moduleFileExtensions": ["js", "json", "ts"],
-  "rootDir": ".",
+  "roots": ["<rootDir>/../src/", "<rootDir>"],
```
Comment (tayormi) [inline test/jest-e2e.json:3]: Codacy found an issue: [Format Error: expected "\n    " ](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692)

File: target/npmlist.json [hunk 2]
```diff
@@ -0,0 +1 @@
...wt":{"version":"4.0.0","dependencies":{"jsonwebtoken":{"version":"8.5.1"},"passport-strategy":{"version":"1.0.0"}}},"passport-local":{"version":"1.0.0","dependencies":{"passport-strategy":{"version":"1.0.0"}}},"redis":{"version":"3.0.2","dependencies":{"denque":{"version":"1.4.1"},"redis-commands":{"version":"1.5.0"},"redis-errors":{"version":"1.2.0"},"redis-parser":{"version":"3.0.0","dependencies":{"redis-errors":{"version":"1.2.0"}}}}},"reflect-metadata":{"version":"0.1.13"},"rimraf":{"version":"3.0.2","dependencies":{"glob":{"version":"7.1.6","dependencies":{"fs.realpath":{"version":"1.0.0"},"inflight":{"version":"1.0.6","dependencies":{"once":{"version":"1.4.0"},"wrappy":{"version":"1.0.2"}}},"inherits":{"version":"2.0.3"},"minimatch":{"version":"3.0.4","dependencies":{"brace-expansion":{"version":"1.1.11","dependencies":{"balanced-match":{"version":"1.0.0"},"concat-map":{"version":"0.0.1"}}}}},"once":{"version":"1.4.0","dependencies":{"wrappy":{"version":"1.0.2"}}},"path-is-absolute":{"version":"1.0.1"}}}}},"rxjs":{"version":"6.5.5","dependencies":{"tslib":{"version":"1.11.1"}}},"swagger-ui-express":{"version":"4.1.4","dependencies":{"swagger-ui-dist":{"version":"3.25.1"}}}}}
```
Comment (tayormi) [inline target/npmlist.json:1]: Codacy found an issue: [Format Error: unexpected "\"name\":\"CodeClanNigeria\",\"version\":\"1.0.0\",\"dependencies\":{\"@hapi/joi\":{\"version\":\"17.1.1\",\"dependencies\":{\"@hapi/address\":{\"version\":\"4.0.1\",\"dependencies\":{\"@hapi/hoek\":{\"version\":\"9.0.4\"}}},\"@hapi/formula\":{\"version\":\"2.0.0\"},\"@hapi/hoek\":{\"version\":\"9.0.4\"},\"@hapi/pinpoint\":{\"version\":\"2.0.0\"},\"@hapi/topo\":{\"version\":\"5.0.0\",\"dependencies\":{\"@hapi/hoek\":{\"version\":\"9.0.4\"}}}}},\"@nestjs/bull\":{\"version\":\"0.1.1\](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692)

**Comment by tayormi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692&bid=18712017)

File: test/jest-e2e.json (hunk 1 above)
Comment (tayormi) [inline test/jest-e2e.json:3]: Codacy found an issue: [Format Error: expected "\n    " ](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692)

**Comment by tayormi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692&bid=18712017)

File: test/jest-e2e.json [hunk 3]
```diff
@@ -1,9 +1,16 @@
 {
   "moduleFileExtensions": ["js", "json", "ts"],
-  "rootDir": ".",
+  "roots": ["<rootDir>/../src", "<rootDir>"],
```
Comment (tayormi) [inline test/jest-e2e.json:3]: Codacy found an issue: [Format Error: expected "\n    " ](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692)

File: test/jest-e2e.json [hunk 4]
```diff
@@ -1,9 +1,16 @@
 {
+  "roots": ["<rootDir>/../src/", "<rootDir>"],
```
Comment (tayormi) [inline test/jest-e2e.json:2]: Codacy found an issue: [Format Error: expected "\n    " ](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692)

**Comment by tayormi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692&bid=18712017)

**Comment by tayormi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692&bid=18712017)

File: test/jest-e2e.json [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
```
Comment (tayormi) [inline test/jest-e2e.json:2]: Codacy found an issue: [Format Error: expected "\n    " ](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692)

**Comment by tayormi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692&bid=18712017)

File: test/jest-e2e.json (hunk 1 above)
Comment (tayormi) [inline test/jest-e2e.json:3]: Codacy found an issue: [Format Error: expected "\n    " ](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692)

Comment (ofuochi) [reply test/jest-e2e.json:3]: This is a minor formatting issue rather than a critical error

**Comment by tayormi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5718692&bid=18712017)

**Pull request merged**